packages/hoppscotch-sh-admin/src/pages/users/index.vue

```diff
@@ -15,132 +15,182 @@
           <HoppButtonSecondary
             outline
             filled
-            :label="t('users.invited_users')"
+            :label="t('users.pending_invites')"
             :to="'/users/invited'"
           />
         </div>
       </div>
       <div class="overflow-x-auto">
-        <div v-if="fetching" class="flex justify-center">
-          <HoppSmartSpinner />
-        </div>
+        <div class="mb-3 flex items-center justify-end">
+          <HoppButtonSecondary
+            outline
+            filled
+            :icon="IconLeft"
+            :disabled="page === 1"
+            @click="changePage(PageDirection.Previous)"
+          />
 
-        <div v-else-if="error">{{ t('users.load_list_error') }}</div>
+          <div class="flex h-full w-10 items-center justify-center">
+            <span>{{ page }}</span>
+          </div>
 
-        <HoppSmartTable v-else-if="usersList.length" :list="usersList">
+          <HoppButtonSecondary
+            outline
+            filled
+            :icon="IconRight"
+            :disabled="page >= totalPages"
+            @click="changePage(PageDirection.Next)"
+          />
+        </div>
+
+        <HoppSmartTable
+          :headings="headings"
+          :list="finalUsersList"
+          :loading="showSpinner"
```

Comment:
There's a slight difference in height compared to the spinner and the row with actual data. We could leverage the `loading-state` slot to implement a custom spinner to fine-tune the dimensions.